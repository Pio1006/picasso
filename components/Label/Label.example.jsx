import React from 'react'
import { storiesOf } from '@storybook/react'

import StoryTeller from '../../.storybook/StoryTeller'
import Label from '.'
import Spacer from '../Spacer'

const stories = storiesOf('Label', module)

const teller = new StoryTeller(
  'Label',
  `Labels are used to describe other topics, incluidng textareas,
  form fields, users, and more. By default, labels are read-only UI elements.
  They are used to surface important information about a topic. Labels may also
  be used to convey status, or used within a group to show selection.`
)
const chapter = teller.addChapter()

function handleDelete () {
  alert('You clicked the delete icon.') // eslint-disable-line no-undef
}

chapter.addSection('Default', null, () => (
  <div>
    <Label label='Javascript' />
  </div>
))

chapter.addSection('Dismissible', null, () => (
  <div>
    <Label label='React JS' onDelete={handleDelete} />
  </div>
))

chapter.addSection('Flat', null, () => (
  <div>
    <Label label='Ember JS' variant='flat' />
  </div>
))

chapter.addSection('Statuses', 'Use these to communicate status.', () => (
  <>
    <Spacer right={1}>
      <Label label={`Yay! It's done!`} variant='success' />
    </Spacer>
    <div>
      <Label label='Nope! Please, try one more time' variant='error' />
    </div>
  </>
))

stories.addWithChapters('Label', teller.toStory())
