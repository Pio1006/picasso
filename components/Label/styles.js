export default ({ palette }) => ({
  flat: {
    backgroundColor: palette.grey[75],
    color: palette.grey[100]
  },
  success: {
    backgroundColor: palette.secondary.light,
    color: palette.secondary.main
  },
  error: {
    backgroundColor: palette.error.light,
    color: palette.error.main
  },

  deleteIcon: {
    display: 'flex',
    justifyContent: 'center',

    color: palette.primary.main,
    fontSize: '0.85em',
    margin: '0 0.5em 0 -0.5em',

    opacity: 0.5,

    '&:hover': {
      opacity: 1,
      color: palette.primary.main
    }
  }
})
