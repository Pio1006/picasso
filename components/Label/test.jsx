import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import Label from './index'

const renderLabel = (props = {}) => {
  return render(<Label {...props} />)
}

afterEach(cleanup)

test('renders default variant', () => {
  const { container } = renderLabel()

  expect(container).toMatchSnapshot()
})

test('renders flat variant', () => {
  const { container } = renderLabel({ variant: 'flat' })

  expect(container).toMatchSnapshot()
})

test('renders success variant', () => {
  const { container } = renderLabel({ variant: 'success' })

  expect(container).toMatchSnapshot()
})

test('renders error variant', () => {
  const { container } = renderLabel({ variant: 'error' })

  expect(container).toMatchSnapshot()
})

describe('dismissable label', () => {
  let onDelete
  let api

  beforeEach(() => {
    onDelete = jest.fn()
    api = renderLabel({ onDelete })
  })
  test('should render dismissable label', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })

  test('should fire onDelete event on dismiss action', () => {
    const { getByTestId } = api
    const deleteIcon = getByTestId('icon-delete')

    fireEvent.click(deleteIcon)
    expect(onDelete).toHaveBeenCalled()
  })
})
