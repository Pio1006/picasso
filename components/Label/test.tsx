import React from 'react'
import { render, fireEvent, cleanup, RenderResult } from 'react-testing-library'

import Label from './index'
import Picasso from '../Picasso'

const renderLabel = (props = {}) => {
  return render(
    <Picasso loadFonts={false}>
      <Label {...props} />
    </Picasso>
  )
}

afterEach(cleanup)

test('renders default variant', () => {
  const { container } = renderLabel()

  expect(container).toMatchSnapshot()
})

test('renders flat variant', () => {
  const { container } = renderLabel({ variant: 'flat' })

  expect(container).toMatchSnapshot()
})

test('renders success variant', () => {
  const { container } = renderLabel({ variant: 'success' })

  expect(container).toMatchSnapshot()
})

test('renders error variant', () => {
  const { container } = renderLabel({ variant: 'error' })

  expect(container).toMatchSnapshot()
})

describe('dismissable label', () => {
  let onDelete: () => void
  let api: RenderResult
  const closeIconTestId = 'icon-delete'

  beforeEach(() => {
    onDelete = jest.fn()
    api = renderLabel({ onDelete, closeIconTestId })
  })
  test('should render dismissable label', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })

  test('should fire onDelete event on dismiss action', () => {
    const { getByTestId } = api
    const deleteIcon = getByTestId(closeIconTestId)

    fireEvent.click(deleteIcon)
    expect(onDelete).toHaveBeenCalled()
  })
})
