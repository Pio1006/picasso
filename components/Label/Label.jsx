import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import CloseIcon from '@material-ui/icons/Close'

import styles from './styles'
import Chip from '../Chip'
import LabelGroup from '../LabelGroup'

const Label = props => {
  const { classes, variant, ...rest } = props

  let rootClass = {
    flat: classes.flat,
    success: classes.success,
    error: classes.error
  }[variant]

  return (
    <Chip
      classes={{ root: rootClass }}
      deleteIcon={
        <CloseIcon className={classes.deleteIcon} data-testid='icon-delete' />
      }
      {...rest}
    />
  )
}

Label.propTypes = {
  label: PropTypes.string,
  onDelete: PropTypes.func,
  variant: PropTypes.oneOf(['flat', 'success', 'error'])
}

Label.defaultProps = {
  label: null,
  onDelete: undefined,
  variant: null
}

Label.Group = LabelGroup

export default withStyles(styles)(Label)
