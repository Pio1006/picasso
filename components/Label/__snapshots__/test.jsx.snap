// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dismissable label should render dismissable label 1`] = `
<div>
  <div
    class="MuiChip-root Chip-root MuiChip-deletable"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-151 Label-deleteIcon-124 MuiChip-deleteIcon-146"
      data-testid="icon-delete"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</div>
`;

exports[`renders default variant 1`] = `
<div>
  <div
    class="MuiChip-root Chip-root"
    role="button"
    tabindex="-1"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</div>
`;

exports[`renders error variant 1`] = `
<div>
  <div
    class="MuiChip-root Chip-root Label-error"
    role="button"
    tabindex="-1"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</div>
`;

exports[`renders flat variant 1`] = `
<div>
  <div
    class="MuiChip-root Chip-root Label-flat"
    role="button"
    tabindex="-1"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</div>
`;

exports[`renders success variant 1`] = `
<div>
  <div
    class="MuiChip-root Chip-root Label-success"
    role="button"
    tabindex="-1"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</div>
`;
